From f3dbbaa2a6c8ddfc22feac62e1fea6310a4fd659 Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Fri, 31 Mar 2023 11:52:25 +0100
Subject: [PATCH] dt-bindings: media: i2c: Replace IMX708 sensor
 binding documentation file

Replace the existing imx708.yaml file with sony,imx708.yaml that follows
the latest devicetree conventions for camera sensors.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 .../devicetree/bindings/media/i2c/imx708.yaml | 119 ----------------
 .../bindings/media/i2c/sony,imx708.yaml       | 128 ++++++++++++++++++
 MAINTAINERS                                   |   2 +-
 3 files changed, 129 insertions(+), 120 deletions(-)
 delete mode 100644 Documentation/devicetree/bindings/media/i2c/imx708.yaml
 create mode 100644 Documentation/devicetree/bindings/media/i2c/sony,imx708.yaml

--- a/Documentation/devicetree/bindings/media/i2c/imx708.yaml
+++ /dev/null
@@ -1,119 +0,0 @@
-# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
-%YAML 1.2
----
-$id: http://devicetree.org/schemas/media/i2c/imx708.yaml#
-$schema: http://devicetree.org/meta-schemas/core.yaml#
-
-title: Sony 1/2.3-Inch 12Mpixel CMOS Digital Image Sensor
-
-maintainers:
-  - Naushir Patuck <naush@raspberypi.com>
-
-description: |-
-  The Sony IMX708 is a 1/2.3-inch CMOS active pixel digital image sensor
-  with an active array size of 4608H x 2592V. It is programmable through
-  I2C interface. The I2C address is fixed to 0x1A as per sensor data sheet.
-  Image data is sent through MIPI CSI-2, which is configured as either 2 or
-  4 data lanes.
-
-properties:
-  compatible:
-    const: sony,imx708
-
-  reg:
-    description: I2C device address
-    maxItems: 1
-
-  clocks:
-    maxItems: 1
-
-  VDIG-supply:
-    description:
-      Digital I/O voltage supply, 1.1 volts
-
-  VANA1-supply:
-    description:
-      Analog1 voltage supply, 2.8 volts
-
-  VANA2-supply:
-    description:
-      Analog2 voltage supply, 1.8 volts
-
-  VDDL-supply:
-    description:
-      Digital core voltage supply, 1.8 volts
-
-  reset-gpios:
-    description: |-
-      Reference to the GPIO connected to the xclr pin, if any.
-      Must be released (set high) after all supplies and INCK are applied.
-
-  # See ../video-interfaces.txt for more details
-  port:
-    type: object
-    properties:
-      endpoint:
-        type: object
-        properties:
-          data-lanes:
-            description: |-
-              The sensor supports either two-lane, or four-lane operation.
-              For two-lane operation the property must be set to <1 2>.
-            items:
-              - const: 1
-              - const: 2
-
-          clock-noncontinuous:
-            type: boolean
-            description: |-
-              MIPI CSI-2 clock is non-continuous if this property is present,
-              otherwise it's continuous.
-
-          link-frequencies:
-            allOf:
-              - $ref: /schemas/types.yaml#/definitions/uint64-array
-            description:
-              Allowed data bus frequencies.
-
-        required:
-          - link-frequencies
-
-required:
-  - compatible
-  - reg
-  - clocks
-  - VANA1-supply
-  - VANA2-supply
-  - VDIG-supply
-  - VDDL-supply
-  - port
-
-additionalProperties: false
-
-examples:
-  - |
-    i2c0 {
-        #address-cells = <1>;
-        #size-cells = <0>;
-
-        imx708: sensor@1a {
-            compatible = "sony,imx708";
-            reg = <0x1a>;
-            clocks = <&imx708_clk>;
-            VANA1-supply = <&imx708_vana1>; /* 1.8v */
-            VANA2-supply = <&imx708_vana2>; /* 2.8v */
-            VDIG-supply = <&imx708_vdig>;   /* 1.1v */
-            VDDL-supply = <&imx708_vddl>;   /* 1.8v */
-
-            port {
-                imx708_0: endpoint {
-                    remote-endpoint = <&csi1_ep>;
-                    data-lanes = <1 2>;
-                    clock-noncontinuous;
-                    link-frequencies = /bits/ 64 <450000000>;
-                };
-            };
-        };
-    };
-
-...
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/sony,imx708.yaml
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/sony,imx708.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Sony 1/2.3-Inch 12Mpixel CMOS Digital Image Sensor
+
+maintainers:
+  - Raspberry Pi Kernel Maintenance <kernel-list@raspberrypi.com>
+
+description: |-
+  The Sony IMX708 is a 1/2.3-inch CMOS active pixel digital image sensor
+  with an active array size of 4608H x 2592V. It is programmable through
+  I2C interface. The I2C address is fixed to 0x1A as per sensor data sheet.
+  Image data is sent through MIPI CSI-2, which is configured as either 2 or
+  4 data lanes.
+
+properties:
+  compatible:
+    const: sony,imx708
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    maxItems: 1
+
+  clock-names:
+    description: Input clock (6 to 27 MHz)
+    items:
+      - const: inck
+
+  vdig-supply:
+    description:
+      Digital I/O voltage supply, 1.1 volts
+
+  vana1-supply:
+    description:
+      Analog1 voltage supply, 2.8 volts
+
+  vana2-supply:
+    description:
+      Analog2 voltage supply, 1.8 volts
+
+  vddl-supply:
+    description:
+      Digital core voltage supply, 1.8 volts
+
+  reset-gpios:
+    description: Sensor reset (XCLR) GPIO
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    description: |
+      Video output port
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            anyOf:
+              - items:
+                  - const: 1
+                  - const: 2
+              - items:
+                  - const: 1
+                  - const: 2
+                  - const: 3
+                  - const: 4
+
+          link-frequencies: true
+
+        required:
+          - data-lanes
+          - link-frequencies
+
+    additionalProperties: false
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - clock-names
+  - vdig-supply
+  - vana1-supply
+  - vana2-supply
+  - vddl-supply
+  - port
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        imx708: camera-sensor@1a {
+            compatible = "sony,imx708";
+            reg = <0x1a>;
+
+            clocks = <&clk 90>;
+            clock-names = "inck";
+
+            vdig-supply = <&camera_vdig>;
+            vana1-supply = <&camera_vana1>;
+            vana2-supply = <&camera_vana2>;
+            vddl-supply = <&camera_vddl>;
+
+            reset-gpios = <&gpio 35 GPIO_ACTIVE_LOW>;
+
+            port {
+                imx708_ep: endpoint {
+                    data-lanes = <1 2>;
+                    link-frequencies = /bits/ 64 <450000000>;
+                    remote-endpoint = <&csi_ep>;
+                };
+            };
+        };
+    };
+...
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -19351,7 +19351,7 @@ M:	Raspberry Pi Kernel Maintenance <kern
 L:	linux-media@vger.kernel.org
 S:	Maintained
 T:	git git://linuxtv.org/media_tree.git
-F:	Documentation/devicetree/bindings/media/i2c/imx708.yaml
+F:	Documentation/devicetree/bindings/media/i2c/sony,imx708.yaml
 F:	drivers/media/i2c/imx708.c
 
 SONY MEMORYSTICK SUBSYSTEM
